File: artifacts/mockup-sandbox/src/components/mockups/engapp/_lib/theme.ts
// Design tokens distilled from the Figma "Feed Card Design" Make project.
// Editorial / brutalist palette: warm beige page, off-white card, true-black
// foreground, and one electric lime accent. Per-POS swatches keep the card
// recognisable across word types.

export const COLORS = {
  page: "#E8E4DC",
  card: "#F5F2EC",
  cardAlt: "#EFEAE0",
  ink: "#0F0F0F",
  inkSoft: "#1A1A1A",
  mute: "#888888",
  muteSoft: "#A8A8A8",
  hairline: "#0F0F0F",
  accent: "#D4FF3D",
  accentSoft: "#EAFFA1",
  warn: "#FFB347",
  pop: "#FF6B8A",
  cool: "#7FD8C9",
  paper: "#FFFFFF",
} as const;

export const POS_COLOR: Record<string, string> = {
  n: COLORS.accent,
  noun: COLORS.accent,
  adj: COLORS.warn,
  adjective: COLORS.warn,
  v: COLORS.pop,
  verb: COLORS.pop,
  adv: COLORS.cool,
  adverb: COLORS.cool,
};

export const FONT_SANS = "Inter, ui-sans-serif, system-ui, -apple-system, sans-serif";
export const FONT_SERIF_TC =
  '"Noto Serif TC", "Source Han Serif TC", "Noto Serif", Georgia, serif';

// CEFR ladder used to render the level chip and stats projection.
export const CEFR_LEVELS = ["A1", "A2", "B1", "B2", "C1", "C2"] as const;
export type CefrLevel = (typeof CEFR_LEVELS)[number];

// Default themes/exams/source-lists used in Browse/Filter and Onboarding.
// Backend has no "list themes" endpoint yet, so we keep a curated list here.
export const THEMES = [
  "nature",
  "emotion",
  "business",
  "travel",
  "loneliness",
  "reflection",
  "city",
  "food",
  "technology",
  "relationships",
] as const;

export const EXAMS = [
  { code: "HKDSE", label: "HKDSE" },
  { code: "TOEIC", label: "TOEIC" },
  { code: "TOEFL", label: "TOEFL" },
  { code: "IELTS", label: "IELTS" },
  { code: "Eiken", label: "Eiken" },
  { code: "Suneung", label: "수능 Suneung" },
] as const;

export const SOURCE_LISTS = [
  "NGSL",
  "NAWL",
  "TSL",
  "BSL",
  "CEFR-J",
] as const;

export const LANGUAGES = [
  { code: "zh-TW", label: "繁體中文（台灣）", english: "Traditional (Taiwan)" },
  { code: "zh-HK", label: "繁體中文（香港）", english: "Traditional (Hong Kong)" },
  { code: "ja", label: "日本語", english: "Japanese" },
  { code: "ko", label: "한국어", english: "Korean" },
  { code: "en", label: "English", english: "English" },
] as const;

export const LEARNING_GOALS = [
  { code: "casual", label: "Casual", subtitle: "no pressure / 隨便看看" },
  { code: "travel", label: "Travel", subtitle: "for trips / 旅行使用" },
  { code: "business", label: "Business", subtitle: "office / 商業英語" },
  { code: "toefl", label: "TOEFL", subtitle: "exam / 托福" },
  { code: "ielts", label: "IELTS", subtitle: "exam / 雅思" },
] as const;

export const DAILY_GOAL_OPTIONS = [3, 5, 10, 20, 50] as const;

File: artifacts/mobile/lib/theme/theme.dart
// Design tokens for EngApp.
//
// "Clean but cute" — pulled back from the strictly editorial look of the
// React mockups. Warm beige page, off-white cards with rounded corners +
// soft drop shadow (not hard hairlines), deep warm charcoal instead of pure
// black, and a friendlier pastel accent palette beside the hero lime.

import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';

class AppColors {
  // Surfaces
  static const Color page = Color(0xFFF3EDE2); // warm cream
  static const Color card = Color(0xFFFAF6EE); // off-white
  static const Color cardAlt = Color(0xFFF1ECDF);
  static const Color paper = Color(0xFFFFFBF3);

  // Ink
  static const Color ink = Color(0xFF1B1A17); // deep warm charcoal
  static const Color inkSoft = Color(0xFF34302A);
  static const Color mute = Color(0xFF8A8073);
  static const Color muteSoft = Color(0xFFC8BFB0);
  static const Color hairline = Color(0xFFE3DCCD);

  // Accent palette
  static const Color accent = Color(0xFFD4FF3D); // hero lime
  static const Color accentSoft = Color(0xFFEBFFA9);
  static const Color peach = Color(0xFFFFC9A4);
  static const Color sage = Color(0xFFB7D3B0);
  static const Color blush = Color(0xFFFFD3D8);
  static const Color butter = Color(0xFFFFE9A8);
  static const Color sky = Color(0xFFB8DCE8);
  static const Color lilac = Color(0xFFD8C7E8);
  static const Color warn = Color(0xFFFFB347);
  static const Color pop = Color(0xFFFF8FA8);
  static const Color cool = Color(0xFF8EDED0);
}

class AppRadii {
  static const double xs = 8;
  static const double sm = 12;
  static const double md = 16;
  static const double lg = 22;
  static const double xl = 28;
  static const double pill = 999;
}

class AppShadows {
  static List<BoxShadow> soft({double y = 8, double blur = 24, double opacity = 0.06}) => [
        BoxShadow(
          color: const Color(0xFF1B1A17).withValues(alpha: opacity),
          offset: Offset(0, y),
          blurRadius: blur,
        ),
      ];

  static List<BoxShadow> sticker = [
        BoxShadow(
          color: const Color(0xFF1B1A17).withValues(alpha: 0.08),
          offset: const Offset(0, 4),
          blurRadius: 0,
        ),
      ];
}

class AppFonts {
  static TextStyle sans({
    double? size,
    FontWeight? weight,
    double? letterSpacing,
    Color? color,
    double? height,
    FontStyle? style,
    TextDecoration? decoration,
  }) {
    return GoogleFonts.inter(
      fontSize: size,
      fontWeight: weight,
      letterSpacing: letterSpacing,
      color: color,
      height: height,
      fontStyle: style,
      decoration: decoration,
    );
  }

  static TextStyle serif({
    double? size,
    FontWeight? weight,
    double? letterSpacing,
    Color? color,
    double? height,
  }) {
    return GoogleFonts.notoSerifTc(
      fontSize: size,
      fontWeight: weight,
      letterSpacing: letterSpacing,
      color: color,
      height: height,
    );
  }

  static TextStyle display({
    double? size,
    FontWeight? weight,
    Color? color,
    double? height,
    double? letterSpacing,
  }) {
    return GoogleFonts.fraunces(
      fontSize: size,
      fontWeight: weight,
      color: color,
      height: height,
      letterSpacing: letterSpacing,
    );
  }
}

/// Small mixed-case label that replaces the all-caps editorial eyebrow.
class SoftLabel extends StatelessWidget {
  final String text;
  final Color? color;
  final double size;
  const SoftLabel(this.text, {super.key, this.color, this.size = 11});

  @override
  Widget build(BuildContext context) {
    return Text(
      text,
      style: AppFonts.sans(
        size: size,
        weight: FontWeight.w500,
        letterSpacing: 0.2,
        color: color ?? AppColors.mute,
      ),
    );
  }
}

class Hairline extends StatelessWidget {
  final double height;
  final Color? color;
  const Hairline({super.key, this.height = 1, this.color});

  @override
  Widget build(BuildContext context) {
    return Container(
      height: height,
      color: color ?? AppColors.hairline,
    );
  }
}

const List<String> cefrLevels = ['A1', 'A2', 'B1', 'B2', 'C1', 'C2'];

const List<String> themesAll = [
  'nature',
  'business',
  'emotion',
  'loneliness',
  'reflection',
  'travel',
  'food',
  'love',
  'time',
  'work',
  'family',
  'art',
];

const List<String> examsAll = [
  'HKDSE',
  'TOEIC',
  'TOEFL',
  'IELTS',
  'Eiken',
  'Suneung',
];

const List<String> sourceListsAll = ['NGSL', 'NAWL', 'TSL', 'BSL', 'CEFR-J'];

const List<int> dailyGoalOptions = [3, 5, 10, 20, 50];

class LearningGoal {
  final String code;
  final String label;
  final String subtitle;
  const LearningGoal(this.code, this.label, this.subtitle);
}

const List<LearningGoal> learningGoals = [
  LearningGoal('casual', 'Casual', '無壓力 · 慢慢學'),
  LearningGoal('travel', 'Travel', '旅遊實用'),
  LearningGoal('business', 'Business', '商務英語'),
  LearningGoal('toefl', 'TOEFL', '托福備考'),
  LearningGoal('ielts', 'IELTS', '雅思備考'),
];

class UiLanguage {
  final String code;
  final String label;
  final String native;
  const UiLanguage(this.code, this.label, this.native);
}

const List<UiLanguage> uiLanguages = [
  UiLanguage('zh-TW', 'Traditional Chinese', '繁體中文（台灣）'),
  UiLanguage('zh-HK', 'Cantonese', '繁體中文（香港）'),
  UiLanguage('ja', 'Japanese', '日本語'),
  UiLanguage('ko', 'Korean', '한국어'),
  UiLanguage('en', 'English', 'English'),
];

Color posColor(String? pos) {
  switch (pos) {
    case 'n':
    case 'noun':
      return AppColors.sage;
    case 'adj':
      return AppColors.butter;
    case 'v':
    case 'verb':
      return AppColors.blush;
    case 'adv':
      return AppColors.sky;
    default:
      return AppColors.sage;
  }
}

/// Rotating pastel for chip-style tags so groups of tags look friendly rather
/// than uniform.
Color tagPastel(int index) {
  const palette = [
    AppColors.peach,
    AppColors.sage,
    AppColors.blush,
    AppColors.butter,
    AppColors.sky,
    AppColors.lilac,
  ];
  return palette[index % palette.length];
}
